workqueue_test.go

```diff
@@ -26,46 +27,46 @@ func TestShouldEnqueueIntoTheUnderlyingWorkQueue(t *testing.T) {
 }
 
 func TestShouldCallRegisteredAddFuncWhenAddEventIsReceived(t *testing.T) {
-	workQueue := kubehandler.NewWorkQueue("WorkqueueTest2")
+	workQueue := kubehandlerv2.NewWorkQueue("WorkqueueTest2")
 	kind := "Foo"
 	addHandlerCalled := make(chan bool, 1)
 	stopChan := make(chan struct{}, 1)
-	workQueue.RegisterAddHandler(kind, func(namespace, name string) error {
+	workQueue.RegisterAddHandler(kind, func(ctx context.Context, namespace, name string) error {
 		addHandlerCalled <- true
 		return nil
 	})
 	workQueue.EnqueueAdd(kind, &appsv1.Deployment{})
-	go workQueue.Run(1, stopChan)
+	go workQueue.Run(context.TODO(), 1)
 	assert.True(t, <-addHandlerCalled)
 	close(stopChan)
 }
 
 func TestShouldCallRegisteredUpdateFuncWhenUpdateEventIsReceived(t *testing.T) {
-	workQueue := kubehandler.NewWorkQueue("WorkqueueTest3")
+	workQueue := kubehandlerv2.NewWorkQueue("WorkqueueTest3")
 	kind := "Foo"
 	updateHandlerCalled := make(chan bool, 1)
 	stopChan := make(chan struct{}, 1)
-	workQueue.RegisterUpdateHandler(kind, func(namespace, name string) error {
+	workQueue.RegisterUpdateHandler(kind, func(ctx context.Context, namespace, name string) error {
 		updateHandlerCalled <- true
 		return nil
 	})
 	workQueue.EnqueueUpdate(kind, &appsv1.Deployment{})
-	go workQueue.Run(1, stopChan)
+	go workQueue.Run(context.TODO(), 1)
 	assert.True(t, <-updateHandlerCalled)
 	close(stopChan)
 }
 
 func TestShouldCallRegisteredDeleteFuncWhenDeleteEventIsReceived(t *testing.T) {
-	workQueue := kubehandler.NewWorkQueue("WorkqueueTest4")
+	workQueue := kubehandlerv2.NewWorkQueue("WorkqueueTest4")
 	kind := "Foo"
 	deleteHandlerCalled := make(chan bool, 1)
 	stopChan := make(chan struct{}, 1)
```

Comment:
here as well

Reply:
Have removed the usage of stopCh and started using context